api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -386,6 +387,26 @@ public RawlsWorkspaceACLUpdateResponseList updateWorkspaceACL(
         (context) -> workspacesApi.updateACL(aclUpdates, workspaceNamespace, firecloudName, false));
   }
 
+  /**
+   * Updates the Access Control List (ACL) for a specified workspace. If Published: Grant Reader
+   * access to all users in the groupEmail If Unpublished: Grant No access to all users in the
+   * groupEmail
+   *
+   * @param workspaceNamespace the Namespace (Terra Billing Project) of the Workspace to modify
+   * @param firecloudName the Terra Name of the Workspace to modify
+   * @param groupEmail the email of the group to update
+   * @return
+   */
+  @Override
+  public void updatePublishWorkspaceACL(
+      String workspaceNamespace, String firecloudName, String groupEmail, boolean publish) {
```

Comment:
Since we want to tightly restrict `groupEmail` to the same value (the RT auth domain group), would do you think about calling `workspaceService.getPublishedWorkspacesGroupEmail()` inside this method instead?

Reply:
I wanted to do this, but since the Workspace Service is also calling it, I was concerned about creating a circular reference. I could call accessTierService.getRegisteredTierOrThrow().getAuthDomainGroupEmail() within this method, but that would result in duplicate code. I'll consider if the  method  getPublishedWorkspacesGroupEmail can be moved to another location to avoid this issue.

Reply:
One think we can do is pass `workspaceService` as the parameter.  When it's WorkspaceService itself calling the method, it can pass `this`.